src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,159 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class Duke {
+    private static final String HORIZONTAL_LINE = "____________________________________________________________";
+    private static final int MAX_TASKS = 100;
+    private static final int TODO_MIN_LENGTH = 6;
+    private static final int TODO_DESCRIPTION_START = 5;
+    private static final String TODO_DESCRIPTION_ERROR = "The todo description cannot be empty";
+    private static final int EVENT_MIN_LENGTH = 7;
+    private static final int EVENT_DESCRIPTION_START = 6;
+    private static final String EVENT_EMPTY_ERROR = "The event must contain an '/at'";
+    private static final String EVENT_DESCRIPTION_ERROR = "The event description cannot be empty";
+    private static final int DEADLINE_MIN_LENGTH = 10;
+    private static final int DEADLINE_DESCRIPTION_START = 9;
+    private static final String DEADLINE_EMPTY_ERROR = "The deadline must contain a '/by'";
+    private static final String DEADLINE_DESCRIPTION_ERROR = "The deadline description cannot be empty";
+    private static final int DONE_MIN_LENGTH = 2;
+    private static final String DONE_DESCRIPTION_ERROR = "The task to be marked as done cannot be empty";
+    private static final String DONE_EXCEED_ERROR = "The task to be marked as done does not exist in the list of tasks";
+    private static final int EVENT_DEADLINE_TIME = 3;
+    private static final String COMMAND_ERROR = "Sorry, I don't understand what you are saying";
```

Comment:
Would it be better if you name your constants with the command words grouped at the front? For example, `DONE_ERROR_EXIT` and `DONE_ERROR_DESCRIPTION`.